[documentation] Use NMT to track native memory leaks (#1917)
reactor/reactor-netty: docs/asciidoc/alloc-metrics.adoc

[width="100%",options="header"]
|=======
| metric name | type | description

> | reactor.netty.bytebuf.allocator.used.heap.memory | Gauge | The number of bytes committed to heap buffer allocator
> | reactor.netty.bytebuf.allocator.used.direct.memory | Gauge | The number of bytes committed to direct buffer allocator

Reviewer: how about "reserved by" rather than "committed to"?

```suggestion
| reactor.netty.bytebuf.allocator.used.heap.memory | Gauge | The number of bytes reserved by heap buffer allocator
| reactor.netty.bytebuf.allocator.used.direct.memory | Gauge | The number of bytes reserved by direct buffer allocator
```

if accepted, please reflect this change in wording in the various other files (javadocs)
